src/hooks/useActiveAccount.tsx

```diff
@@ -73,60 +73,51 @@ export const ActiveAccountContextProvider = ({
   /**
    * Initial setting of activeAccount
    */
-  useEffect(() => {
-    if (
-      accountsResult.isFetched &&
-      !accountsResult.isLoading &&
-      accountsWithProduct.length < 1
-    ) {
-      setIsSettled(true);
-    }
-
+  const activeAccount = useMemo<ActiveAccountProps>(() => {
     if (
       !userId || // require user id before reading/writing to storage
       storedAccountIdResult.isLoading || // wait for async storage result
-      activeAccount?.account?.id || // activeAccount already set
       accountsWithProduct.length < 1 // no valid accounts found server side
     ) {
-      return;
+      return {};
     }
 
     const accountToSelect =
-      accountIdToSelect ?? storedAccountIdResult.data ?? undefined;
+      selectedId ?? storedAccountIdResult.data ?? undefined;
 
     const selectedAccount =
       getValidAccount(accountsWithProduct, accountToSelect) ??
       accountsWithProduct[0];
 
-    setActiveAccount({
+    return {
       account: selectedAccount,
       accountHeaders: {
         'LifeOmic-Account': selectedAccount.id,
       },
       trialExpired: getTrialExpired(selectedAccount),
-    });
-    setStoredAccountId(selectedAccount.id);
-    setIsSettled(true);
+    };
   }, [
     accountsWithProduct,
-    activeAccount?.account?.id,
-    accountIdToSelect,
-    setStoredAccountId,
+    selectedId,
     storedAccountIdResult.data,
     storedAccountIdResult.isLoading,
-    accountsResult.isFetched,
-    accountsResult.isLoading,
     userId,
   ]);
 
+  useEffect(() => {
+    if (activeAccount?.account?.id) {
+      setStoredAccountId(activeAccount.account.id);
```

Comment:
It seems like this could write to async storage more often than we mean to.  You might want to make the dependency on line 111 `activeAccount?.account?.id` instead.